Update GCC RISC-V QEMU project to support new RISC-V port and vector mode (#780)
FreeRTOS/FreeRTOS: FreeRTOS/Demo/RISC-V_RV32_QEMU_VIRT_GCC/Readme.md

# Emulating generic RISC-V 32bit machine on QEMU

## Requirements

> 1. GNU RISC-V toolchains (tested on pre-build Sifive GNU Embedded Toolchain — v2020.12.8)

Reviewer: ```suggestion
1. GNU RISC-V toolchains (tested on pre-built Sifive GNU Embedded Toolchain — v2020.12.8)
```

  - https://www.sifive.com/software

> 1. qemu-riscv32-system  (tested on pre-build Sifive QEMU — v2020.08.1)

Reviewer: ```suggestion
1. qemu-riscv32-system  (tested on pre-built Sifive QEMU — v2020.08.1)
```

Author: Done

  - https://www.sifive.com/software
1. Linux OS (tested on Ubuntu 20.04.3 LTS)

## How to build

Add path of toolchain that is described above section, such as:

```
$ export PATH="/YOUR_PATH/riscv64-unknown-elf/bin:${PATH}"
```

For release build:

```
$ make -C build/gcc/
```

For debug build:

```
$ make -C build/gcc/ DEBUG=1
```

> For clean build artifacts:

Reviewer: ```suggestion
To clean build artifacts:
```

Author: Done

```
$ make -C build/gcc/ clean
```

> If success to build, executable file RTOSDemo.elf in ./build/gcc/output directory.

Reviewer: ```suggestion
If the build was successful, the RTOSDemo.elf executable will be located in the build/gcc/output directory.
```

Author: Done

## How to run

```
$ qemu-system-riscv32 -nographic -machine virt -net none \
  -chardev stdio,id=con,mux=on -serial chardev:con \
  -mon chardev=con,mode=readline -bios none \
  -smp 4 -kernel ./build/gcc/output/RTOSDemo.elf
```

## How to debug with gdb

Append -s and -S options to the previous qemu command.

- -s: enable to attach gdb to QEMU at port 1234
- -S: start and halted CPU (wait for attach from gdb)

> This is just recommend to use 'debug build' for more efficient debugging.

Reviewer: ```suggestion
It is recommended to use the 'debug build' so that gdb can automatically map symbols.
```

Author: Done
